mlflow/utils/requirements_utils.py

```diff
@@ -409,3 +424,41 @@ def _get_pinned_requirement(package, version=None, module=None):
             version = version_raw
 
     return f"{package}=={version}"
+
+
+class _MismatchedPackageInfo(NamedTuple):
+    package_name: str
+    installed_version: Optional[str]
+    requirement: str
+
+    def __str__(self):
+        current_status = self.installed_version if self.installed_version else "uninstalled"
+        return f"{self.package_name} (current: {current_status}, required: {self.requirement})"
+
+
+def _check_requirement_satisfied(requirement_str):
+    """
+    Returns None if the current python environment satisfies the given requirement.
+    Otherwise, returns an instance of `_MismatchedPackageInfo`.
+    """
+    _init_packages_to_modules_map()
+    req = pkg_resources.Requirement.parse(requirement_str)
+    pkg_name = req.name
+
+    try:
+        installed_version = _get_installed_version(pkg_name, _PACKAGES_TO_MODULES.get(pkg_name))
+    except ModuleNotFoundError:
+        return _MismatchedPackageInfo(
+            package_name=pkg_name,
+            installed_version=None,
+            requirement=requirement_str,
+        )
+
+    if req.specifier and not req.specifier.contains(installed_version):
```

Comment:
Do we have a test for a requirement with version specifiers?

Reply:
addressed.